### PR TITLE
Adds Power Spectrum example for a timeseries 

### PR DESCRIPTION
**Pull request by yashkgp**
Issue #2483 
Here is an example : 
```python
import sunpy.timeseries
import sunpy.data.sample
import matplotlib.pyplot as plt
import pandas as pd 
from scipy import signal
ts = sunpy.timeseries.TimeSeries(sunpy.data.sample.GOES_XRS_TIMESERIES)
k,l=ts.power_spectra()
plt.semilogy(k[0], l[0])
plt.show()
```
image

Apart from adding the unit tests,is anything else needed for this method?

### CODE REVIEW

File: sunpy/timeseries/timeseriesbase.py [hunk 1]
```diff
@@ -524,6 +525,35 @@ def _sanitize_metadata(self, **kwargs):
...
+            Power spectral density or power spectrum of all the columns in
+            the TimeSeries.
+        """
+        Freqs = []
+        Spectra = []
+        for col in self.columns:
+            data_ = self.data[col]
+            freq, spectra = signal.periodogram(data_, fs)
+            Freqs.append(freq)
+            Spectra.append(spectra)
```
Comment (dpshelio) [inline sunpy/timeseries/timeseriesbase.py]:
```python
from functools import partial
periodogram_fs = partial(signal.periodogram, fs=fs)
Freqs, Spectra = zip(*map(periodogram_fs, self.data))
```
This will remove all the loop, it will produce tuples instead of lists though.

File: sunpy/timeseries/timeseriesbase.py [hunk 2]
```diff
@@ -524,6 +525,35 @@ def _sanitize_metadata(self, **kwargs):
...
+
+        Parameter
+        ----------
+        fs : `float, optional`
+            Sampling frequency of the time series in units of Hz.
+            Defaults to 1.0.
+
+        Returns
+        -------
+        Freqs : array of arrays
```
Comment (dpshelio) [inline sunpy/timeseries/timeseriesbase.py]: You are not returning `array` but a `list` of arrays.

File: sunpy/timeseries/timeseriesbase.py [hunk 3]
```diff
@@ -524,6 +525,35 @@ def _sanitize_metadata(self, **kwargs):
         redundant_cols = list(set(self.meta.columns) - set(self.columns))
         self.meta._remove_columns(redundant_cols)
 
+    def power_spectra(self, fs = 1):
+        """
+        Estimate power spectral density of the TimeSeries
+
+        Parameter
+        ----------
+        fs : `float, optional`
```
Comment (dpshelio) [inline sunpy/timeseries/timeseriesbase.py]: optional should be outside the backtick.

**Comment by nabobalis** [comment]
Hey @yashkgp, sorry for the late reply.

We aren't sure we want to add methods on to time series right now. It sparked quite a discussion about methods in various parts of SunPy. That shake the foundation of SunPy itself. 

What we want from this now is if you could turn it into an example we could add to our [Gallery](http://docs.sunpy.org/en/stable/generated/gallery/index.html)? If you have a [look at the other examples here](https://github.com/sunpy/sunpy/tree/master/examples) for how we format them!

**Comment by yashkgp** [comment]
Sure @nabobalis Not a Problem .

**Comment by yashkgp** [comment]
Hey @nabobalis Could you give a review.

File: examples/power_spectra_example.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
+"""
+===============================
+POWER SPECTRUM OF A TIMESERIES
```
Comment (nabobalis) [inline examples/power_spectra_example.py]: Not all caps, just normal case.

File: examples/power_spectra_example.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+===============================
+
+An example showing how to estimate power spectrum of a TimeSeries
```
Comment (nabobalis) [inline examples/power_spectra_example.py]: I think a `the` should between `estimate` and `power`. Full stop at the end.

File: examples/power_spectra_example.py [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+import pandas as pd 
+from scipy import signal
+
+import sunpy.timeseries
+from sunpy.data.sample import GOES_XRS_TIMESERIES
+
+###############################################################################
+# Let's first load a TimeSeries from sample data
+
+ts = sunpy.timeseries.TimeSeries(GOES_XRS_TIMESERIES)
```
Comment (nabobalis) [inline examples/power_spectra_example.py]: You might want to describe the data you will working on to give whoever reads this an idea of what the input is.

File: examples/power_spectra_example.py [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+###############################################################################
+# We now use scipy's periodogram to estimate the power spectra of the first
```
Comment (nabobalis) [inline examples/power_spectra_example.py]: You want to explain why you use the periodogram instead of say the FFT. Also we probably want to mention the limitations or requirements  of using these methods. We would like people not to assume they can use these methods on any signal.  

I think there is one in Astropy as well now. But I am unsure of what version.

File: examples/power_spectra_example.py [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+# Let's first load a TimeSeries from sample data
+
+ts = sunpy.timeseries.TimeSeries(GOES_XRS_TIMESERIES)
+
+###############################################################################
+# We now use scipy's periodogram to estimate the power spectra of the first 
+# column of the Timeseries
+
+cols=ts.columns
+freq, spectra = signal.periodogram(ts.data[cols[0]], fs=1)
```
Comment (nabobalis) [inline examples/power_spectra_example.py]: Is `fs=1` correct for the data? 

Also I wonder if we could use the false alarm probability in this example as well.

File: examples/power_spectra_example.py [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+plt.semilogy(freq, spectra)
+plt.title('Power Spectrum of {}'.format(cols[0]))
+plt.ylabel('Power Spectral Density')
+plt.xlabel('Frequency(Hz)')
+plt.show()
+
+
+
+
+
```
Comment (nabobalis) [inline examples/power_spectra_example.py]: All these extra lines need to go.

File: examples/power_spectra_example.py [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+cols=ts.columns
+freq, spectra = signal.periodogram(ts.data[cols[0]], fs=1)
+
+###############################################################################
+# Plot the power spectrum
+
+plt.semilogy(freq, spectra)
+plt.title('Power Spectrum of {}'.format(cols[0]))
+plt.ylabel('Power Spectral Density')
```
Comment (nabobalis) [inline examples/power_spectra_example.py]: Missing the unit

File: examples/power_spectra_example.py (hunk 8 above)
Comment (yashkgp) [reply examples/power_spectra_example.py]: Yeah , this feature is covered in Astropy's periodogram . Should we switch over to Astropy's periodogram ?

Comment (nabobalis) [reply examples/power_spectra_example.py]: If it works out regarding the version of Astropy it is supported in.

File: examples/power_spectra_example.py (hunk 7 above)
Comment (yashkgp) [reply examples/power_spectra_example.py]: Yeah , just discovered that . Astropy's Periodogram has some amazing functionalities . We should probably switch over to Astropy's Periodogram .

**Comment by yashkgp** [comment]
Hey @nabobalis , Just discovered that Astropy's Periodogram is much more powerful than scipy's Periodogram . They have full fledged tutorial dedicated to Periodogram [http://docs.astropy.org/en/stable/stats/lombscargle.html#](url).It has various functionalities like frequency grid spacing, false alarm probabilities, normalization etc.We should definitely switch over to Astrpoy's peiodogram . What do you say @nabobalis ?

File: examples/power_spectra_example.py [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+###############################################################################
+# Let's first load a TimeSeries from sample data
+
+ts = sunpy.timeseries.TimeSeries(GOES_XRS_TIMESERIES)
+
+###############################################################################
+# We now use scipy's periodogram to estimate the power spectra of the first 
+# column of the Timeseries
+
+cols=ts.columns
```
Comment (dpshelio) [inline examples/power_spectra_example.py]: Since probably these examples need to be somehow meaningful I would do the following instead of using a variable called cols:
```python
xrays_short = ts.columns[0]
```
and change `cols[0]` for `xrays_short` everywhere. That makes the code more readable.

File: examples/power_spectra_example.py (hunk 8 above)
Comment (yashkgp) [reply examples/power_spectra_example.py]: Yeah , it's there in Astropy 3.0

Comment (nabobalis) [reply examples/power_spectra_example.py]: Well we still have to support Astropy 2 until our next release.

**Comment by yashkgp** [comment]
Hey @nabobalis @dpshelio , I tried switching to Astropy's version of periodogram (yeah, its supported in Astropy 2 ) but its throwing a ValueError
`TypeError: ufunc true_divide cannot use operands with types dtype('float64') and dtype('<m8[ns]')`
It seems like Astropy's periodogram does accept Pandas Timeseries object . One possible solution might to use Timedelta and convert `dtype('<m8[ns]')` into float . Is there any easy way to solve this problem ?

File: examples/power_spectra_example.py (hunk 11 above)
Comment (yashkgp) [reply examples/power_spectra_example.py]: Done .

File: examples/power_spectra_example.py (hunk 5 above)
Comment (yashkgp) [reply examples/power_spectra_example.py]: Done.

File: examples/power_spectra_example.py (hunk 4 above)
Comment (yashkgp) [reply examples/power_spectra_example.py]: Done.

File: examples/power_spectra_example.py (hunk 6 above)
Comment (yashkgp) [reply examples/power_spectra_example.py]: I have added a brief description about the input data now  . Is it okay ?

**Comment by nabobalis** [comment]
If you can't get the astropy one to work, It would just be worth noting that its available and linking to it in the example.

File: examples/power_spectra_example.py [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+import sunpy.timeseries
+from sunpy.data.sample import GOES_XRS_TIMESERIES
+
+###############################################################################
+# Let's first load a TimeSeries from sample data.
+
+ts = sunpy.timeseries.TimeSeries(GOES_XRS_TIMESERIES)
+# Data is taken from GOES X-Ray Sensor(XRS), which observes full-disk-integrated
+# solar flux in two broadband channels: 1--8 angstrom (long); and
+# 0.5--4 angstrom (short)
```
Comment (nabobalis) [inline examples/power_spectra_example.py]: This should be moved above the python code. 

But what I wanted here was not what is GOES, it was can we use signal analysis on this data. Is it evenly spaced in time, whats its time step, are there any problems that might mean that signal analysis would be tricky etc. We don't want to give the impression you can just singal analysis any signal.

File: examples/power_spectra_example.py [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+
+###############################################################################
+# We now use scipy's periodogram to estimate the power spectra of the first 
+# column of the Timeseries
```
Comment (nabobalis) [inline examples/power_spectra_example.py]: Make a link to the astropy version here, saying it's available instead.

File: examples/power_spectra_example.py [hunk 14]
```diff
@@ -17,23 +17,26 @@
...
+# 0.5--4 angstrom (short)
 
 ###############################################################################
 # We now use scipy's periodogram to estimate the power spectra of the first 
 # column of the Timeseries
 
-cols=ts.columns
-freq, spectra = signal.periodogram(ts.data[cols[0]], fs=1)
+xray_short = ts.columns[0]
+freq, spectra = signal.periodogram(ts.data[xray_short], fs=1)
```
Comment (dpshelio) [inline examples/power_spectra_example.py]: I've looked in detail into the data and I don't believe this is correct. `fs=1` means the cadence of the data in Hz, the data seems to not be completely evenly sampled!!
```python
>>> sample = np.array([(x - y).total_seconds() for x, y in zip(ts.index[1:], ts.index)])
>>> sample.min(), sample.max(), sample.mean(), sample.std()
  (2.0430000000000001, 14.337000000000002, 2.0485031771623676, 0.077912702228747382)
```
I don't know how there's a 14.33 seconds gap!!! If you plot that you see it happens twice! This means that this analysis would be wrong all together!

In case the cadence would be 2.043, then I think the `fs` should be `1/2.043 s = 0.48 Hz`

File: examples/power_spectra_example.py [hunk 15]
```diff
@@ -17,23 +17,26 @@
...
 
 ###############################################################################
 # Plot the power spectrum
 
 plt.semilogy(freq, spectra)
-plt.title('Power Spectrum of {}'.format(cols[0]))
-plt.ylabel('Power Spectral Density')
+plt.title('Power Spectrum of {}'.format(xray_short))
+plt.ylabel('Power Spectral Density(({})**2/Hz)'.format(ts.units[xray_short]))
```
Comment (dpshelio) [inline examples/power_spectra_example.py]: You could use units here as:
```python
import astropy.units as u
plt.ylabel('Power Spectral Density ({:LaTeX})'.format(ts.units[xray_short] ** 2 / u.Hz))
```
Also, put a space between the `Density` and the `()`

File: examples/power_spectra_example.py [hunk 16]
```diff
@@ -17,23 +17,26 @@
...
 plt.xlabel('Frequency(Hz)')
```
Comment (dpshelio) [inline examples/power_spectra_example.py]: Also space between the title and the units.

File: examples/power_spectra_example.py (hunk 15 above)
Comment (dpshelio) [reply examples/power_spectra_example.py]: I believe square brackets for the units are more common: `[Hz]` instead of `(Hz)`

File: examples/power_spectra_example.py (hunk 14 above)
Comment (yashkgp) [reply examples/power_spectra_example.py]: Should I clean the data by removing those 2 outliers ?

File: examples/power_spectra_example.py (hunk 15 above)
Comment (yashkgp) [reply examples/power_spectra_example.py]: Yeah, this is much cleaner.

File: examples/power_spectra_example.py (hunk 12 above)
Comment (yashkgp) [reply examples/power_spectra_example.py]: Data is not completely evenly sampled,it has 2 outliers .Should I mention this?

File: examples/power_spectra_example.py (hunk 14 above)
Comment (yashkgp) [reply examples/power_spectra_example.py]: Should I just change `fs` to 0.48 and notify the users about the irregularities in data or should I change the input data ?

**Comment by yashkgp** [comment]
Hey @nabobalis @dpshelio , I have changed the input data, as the previous one was not evenly sampled . I have switched to `RHESSI_TIMESERIES` , which is a evenly sampled timeseries with a time step of 4 sec. Could you please give it a review?

File: examples/power_spectra_example.py [hunk 17]
```diff
@@ -0,0 +1,43 @@
...
+An example showing how to estimate the power spectrum of a TimeSeries.
+"""
+
+##############################################################################
+# Start by importing the necessary modules.
+
+import matplotlib.pyplot as plt
+import pandas as pd 
+from scipy import signal
+
```
Comment (nabobalis) [inline examples/power_spectra_example.py]: I would remove this blank line.

File: examples/power_spectra_example.py [hunk 18]
```diff
@@ -0,0 +1,43 @@
...
+import astropy.units as u
+
+import sunpy.timeseries
+from sunpy.data.sample import RHESSI_TIMESERIES
+
+###############################################################################
+# Let's first load a TimeSeries from sample data.
+
```
Comment (nabobalis) [inline examples/power_spectra_example.py]: Remove this line.

File: examples/power_spectra_example.py [hunk 19]
```diff
@@ -0,0 +1,43 @@
...
+# Input data contains 9 columns, which are evenly sampled with a time
+# step of 4 seconds.
+ts = sunpy.timeseries.TimeSeries(RHESSI_TIMESERIES)
+
+###############################################################################
+# We now use scipy's periodogram to estimate the power spectra of the first
```
Comment (nabobalis) [inline examples/power_spectra_example.py]: SciPy, also what is the first column?

File: examples/power_spectra_example.py [hunk 20]
```diff
@@ -0,0 +1,43 @@
...
+# Let's first load a TimeSeries from sample data.
+
+# Input data contains 9 columns, which are evenly sampled with a time
+# step of 4 seconds.
+ts = sunpy.timeseries.TimeSeries(RHESSI_TIMESERIES)
+
+###############################################################################
+# We now use scipy's periodogram to estimate the power spectra of the first 
+# column of the Timeseries. Astropy's Lomb-Scargle Periodograms can be used
+# instead of scipy's periodogram.
```
Comment (nabobalis) [inline examples/power_spectra_example.py]: SciPy

File: examples/power_spectra_example.py [hunk 21]
```diff
@@ -0,0 +1,43 @@
...
+###############################################################################
+# Let's first load a TimeSeries from sample data.
+
+# Input data contains 9 columns, which are evenly sampled with a time
+# step of 4 seconds.
+ts = sunpy.timeseries.TimeSeries(RHESSI_TIMESERIES)
+
+###############################################################################
+# We now use scipy's periodogram to estimate the power spectra of the first 
+# column of the Timeseries. Astropy's Lomb-Scargle Periodograms can be used
```
Comment (nabobalis) [inline examples/power_spectra_example.py]: Link to Astropy's documentation for it.

File: examples/power_spectra_example.py [hunk 22]
```diff
@@ -0,0 +1,43 @@
...
+# instead of scipy's periodogram.
+
+x_ray = ts.columns[0]
+# The suitable value for fs would be 0.25 Hz as the time step is 4 s.
+freq, spectra = signal.periodogram(ts.data[xray_short], fs=0.25)
+###############################################################################
```
Comment (nabobalis) [inline examples/power_spectra_example.py]: Blank line here above the `####...`

File: examples/power_spectra_example.py [hunk 23]
```diff
@@ -0,0 +1,43 @@
...
+ts = sunpy.timeseries.TimeSeries(RHESSI_TIMESERIES)
+
+###############################################################################
+# We now use scipy's periodogram to estimate the power spectra of the first 
+# column of the Timeseries. Astropy's Lomb-Scargle Periodograms can be used
+# instead of scipy's periodogram.
+
+x_ray = ts.columns[0]
+# The suitable value for fs would be 0.25 Hz as the time step is 4 s.
+freq, spectra = signal.periodogram(ts.data[xray_short], fs=0.25)
```
Comment (nabobalis) [inline examples/power_spectra_example.py]: x_ray

File: examples/power_spectra_example.py [hunk 24]
```diff
@@ -0,0 +1,43 @@
...
+###############################################################################
+# Plot the power spectrum
+
+plt.semilogy(freq, spectra)
+plt.title('Power Spectrum of {}'.format(xray_short))
```
Comment (nabobalis) [inline examples/power_spectra_example.py]: x_ray

File: examples/power_spectra_example.py (hunk 23 above)
Comment (yashkgp) [reply examples/power_spectra_example.py]: Sorry @nabobalis ,had updated in my local branch , but somehow forgot to push it.

File: examples/power_spectra_example.py (hunk 22 above)
Comment (yashkgp) [reply examples/power_spectra_example.py]: Done

File: examples/power_spectra_example.py [hunk 25]
```diff
@@ -0,0 +1,44 @@
...
+import astropy.units as u
+
+import sunpy.timeseries
+from sunpy.data.sample import RHESSI_TIMESERIES
+
+###############################################################################
+# Let's first load a TimeSeries from sample data.
+# Input data contains 9 columns, which are evenly sampled with a time
+# step of 4 seconds. So, the data is suitable for signal analysis.
+ts = sunpy.timeseries.TimeSeries(RHESSI_TIMESERIES)
```
Comment (nabobalis) [inline examples/power_spectra_example.py]: Line break above here just to be consistent.

File: examples/power_spectra_example.py [hunk 26]
```diff
@@ -0,0 +1,44 @@
...
+import matplotlib.pyplot as plt
+import pandas as pd 
+from scipy import signal
+import astropy.units as u
+
+import sunpy.timeseries
+from sunpy.data.sample import RHESSI_TIMESERIES
+
+###############################################################################
+# Let's first load a TimeSeries from sample data.
```
Comment (nabobalis) [inline examples/power_spectra_example.py]: `Lets first load a RHESSI TimeSeries from SunPy's sample data`

File: examples/power_spectra_example.py [hunk 27]
```diff
@@ -0,0 +1,44 @@
...
+# Input data contains 9 columns, which are evenly sampled with a time
```
Comment (nabobalis) [inline examples/power_spectra_example.py]: `This data contains`

File: examples/power_spectra_example.py [hunk 28]
```diff
@@ -0,0 +1,44 @@
...
+# step of 4 seconds. So, the data is suitable for signal analysis.
```
Comment (nabobalis) [inline examples/power_spectra_example.py]: Remove the last sentence.

File: examples/power_spectra_example.py [hunk 29]
```diff
@@ -0,0 +1,44 @@
...
+ts = sunpy.timeseries.TimeSeries(RHESSI_TIMESERIES)
+
+###############################################################################
+# We now use SciPy's periodogram to estimate the power spectra of the first
```
Comment (nabobalis) [inline examples/power_spectra_example.py]: I think that `~scipy.signal.lombscargle` should work as a link to the documentation

File: examples/power_spectra_example.py [hunk 30]
```diff
@@ -0,0 +1,44 @@
...
+# Let's first load a TimeSeries from sample data.
+# Input data contains 9 columns, which are evenly sampled with a time
+# step of 4 seconds. So, the data is suitable for signal analysis.
+ts = sunpy.timeseries.TimeSeries(RHESSI_TIMESERIES)
+
+###############################################################################
+# We now use SciPy's periodogram to estimate the power spectra of the first 
+# column of the Timeseries. The first column contains X-Ray emmisions in the range 
+# of 3-6 keV. Astropy's Lomb-Scargle Periodograms 
+# (http://docs.astropy.org/en/v2.0.5/stats/lombscargle.html) can be used
```
Comment (nabobalis) [inline examples/power_spectra_example.py]: Try `~astropy.stats.LombScargle` instead. Sphinx should link to it since we have that setup. 

I would rephrase the last sentence as well. `An alterantive version is ....` or something.

File: examples/power_spectra_example.py (hunk 29 above)
Comment (yashkgp) [reply examples/power_spectra_example.py]: Ohh! Thats quite useful. Thanks a lot @nabobalis

File: examples/power_spectra_example.py [hunk 31]
```diff
@@ -0,0 +1,44 @@
...
+===============================
+
+An example showing how to estimate the power spectrum of a TimeSeries.
+"""
+
+##############################################################################
+# Start by importing the necessary modules.
+
+import matplotlib.pyplot as plt
+import pandas as pd
```
Comment (dpshelio) [inline examples/power_spectra_example.py]: pandas is not used

File: examples/power_spectra_example.py [hunk 32]
```diff
@@ -0,0 +1,44 @@
...
+
+###############################################################################
+# Let's first load a RHESSI TimeSeries from SunPy's sample data.
+# This data contains 9 columns, which are evenly sampled with a time step of 4 
+# seconds. 
+
+ts = sunpy.timeseries.TimeSeries(RHESSI_TIMESERIES)
+
+###############################################################################
+# We now use SciPy's periodogram(~scipy.signal.lombscargle) to estimate the power
```
Comment (dpshelio) [inline examples/power_spectra_example.py]: Wait... `signal.periodogram` is the same than `signal.lombscargle`? If so, then you could have used the previous series that was not evenly spaced.

Comment (yashkgp) [reply examples/power_spectra_example.py]: @dpshelio Its not the same , I'll just change it in the next commit .
Sorry @dpshelio , I am not that acquainted with different types of periodogram. Should I use a normal `signal.periodogram` or a `signal.lombscargle`?

Comment (yashkgp) [reply examples/power_spectra_example.py]: Sorry @dpshelio , I am not that well acquainted with lomb scargle periodogram . Is there any advantage of using `signal.lombscargle` over `signal.periodogram`?

Comment (nabobalis) [reply examples/power_spectra_example.py]: They are not the same, that should point to `scipy.signal.periodogram` instead.

**Comment by nabobalis** [comment]
Odd how the sphinx interlink hasn't worked. I wonder what is wrong.

File: examples/power_spectra_example.py [hunk 33]
```diff
@@ -0,0 +1,43 @@
...
+
+###############################################################################
+# Let's first load a RHESSI TimeSeries from SunPy's sample data.
+# This data contains 9 columns, which are evenly sampled with a time step of 4 
+# seconds. 
+
+ts = sunpy.timeseries.TimeSeries(RHESSI_TIMESERIES)
+
+###############################################################################
+# We now use SciPy's periodogram(~scipy.signal.periodogram) to estimate the power
```
Comment (nabobalis) [inline examples/power_spectra_example.py]: There needs to be ` on each side for these links to work.

File: examples/power_spectra_example.py [hunk 34]
```diff
@@ -0,0 +1,43 @@
...
+# This data contains 9 columns, which are evenly sampled with a time step of 4 
+# seconds. 
+
+ts = sunpy.timeseries.TimeSeries(RHESSI_TIMESERIES)
+
+###############################################################################
+# We now use SciPy's periodogram(~scipy.signal.periodogram) to estimate the power 
+# spectra of the first column of the Timeseries. The first column contains X-Ray 
+# emmisions in the range of 3-6 keV. An alterantive version is Astropy's 
+# Lomb-Scargle Periodograms(~astropy.stats.LombScargle).
```
Comment (nabobalis) [inline examples/power_spectra_example.py]: There needs to be ` on each side for these links to work.

**Comment by yashkgp** [comment]
Sorry for the late update , had my exams going on.

File: examples/power_spectra_example.py [hunk 35]
```diff
@@ -0,0 +1,43 @@
...
+
+###############################################################################
+# Let's first load a RHESSI TimeSeries from SunPy's sample data.
+# This data contains 9 columns, which are evenly sampled with a time step of 4 
+# seconds. 
+
+ts = sunpy.timeseries.TimeSeries(RHESSI_TIMESERIES)
+
+###############################################################################
+# We now use SciPy's periodogram(`~scipy.signal.periodogram`) to estimate the
```
Comment (nabobalis) [inline examples/power_spectra_example.py]: Looking at the final result, i think we want this to be `SciPy's ~scipy.signal.periodogram`

File: examples/power_spectra_example.py [hunk 36]
```diff
@@ -0,0 +1,43 @@
...
+# This data contains 9 columns, which are evenly sampled with a time step of 4 
+# seconds. 
+
+ts = sunpy.timeseries.TimeSeries(RHESSI_TIMESERIES)
+
+###############################################################################
+# We now use SciPy's periodogram(`~scipy.signal.periodogram`) to estimate the 
+# power spectra of the first column of the Timeseries. The first column contains
+# X-Ray emmisions in the range of 3-6 keV. An alterantive version is Astropy's 
+# Lomb-Scargle Periodograms(`~astropy.stats.LombScargle`).
```
Comment (nabobalis) [inline examples/power_spectra_example.py]: Similar here. `Astropy's  ~astropy.stats.LombScargle`.

**Comment by nabobalis** [comment]
Thanks @yashkgp!
